ui/src/components/Treasury.jsx

```diff
@@ -78,7 +71,13 @@ function VaultList() {
     return <Redirect to={redirect} />;
   }
 
-  const vaultsList = Object.entries(vaults);
+  if (!approved) {
```

Comment:
I've just grouped all the JSX returns in the same place so it's a little more intuitive as to what's going on in the rendering, just seems more readable to me but it's subjective :)